Vulkan-Pipeline-Assistant/mainwindow.cpp

```diff
@@ -119,19 +143,57 @@ void MainWindow::MakeShaderBlock(QWidget* parent, QString labelStr) {
 
 // TODO expandable vertex inputs (max 16, although track for multiple such as with mat4)
 QWidget* MainWindow::MakeVertexInputBlock() {
+    QMap<QString, VkPrimitiveTopology> topologies;
+    topologies.insert("Point List", VK_PRIMITIVE_TOPOLOGY_POINT_LIST);
+    topologies.insert("Line List", VK_PRIMITIVE_TOPOLOGY_LINE_LIST);
+    topologies.insert("Line Strip", VK_PRIMITIVE_TOPOLOGY_LINE_STRIP);
+    topologies.insert("Triangle List", VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST);
+    topologies.insert("Triangle Strip", VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP);
+    topologies.insert("Triangle Fan", VK_PRIMITIVE_TOPOLOGY_TRIANGLE_FAN);
+    topologies.insert("Line List With Adjacency", VK_PRIMITIVE_TOPOLOGY_LINE_LIST_WITH_ADJACENCY);
+    topologies.insert("Line Strip With Adjacency", VK_PRIMITIVE_TOPOLOGY_LINE_STRIP_WITH_ADJACENCY);
+    topologies.insert("Triangle List With Adjacency", VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST_WITH_ADJACENCY);
+    topologies.insert("Triangle Strip With Adjacency", VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP_WITH_ADJACENCY);
+    topologies.insert("Patch List", VK_PRIMITIVE_TOPOLOGY_PATCH_LIST);
+
     QWidget* parent = new QWidget(m_rightTopContainer);
     QLabel* topologyLabel = new QLabel("Topology", parent);
-    QComboBox* topologyBox = MakeComboBox(parent, {
-        "Point List", "Line List", "Line Strip", "Triangle List", "Triangle Strip", "Trangle Fan",
-        "Line List With Adjacency", "Line Strip With Adjacency", "Triangle List WIth Adjacency",
-        "Triangle Strip With Adjacency", "Patch List"});
+    QComboBox* topologyBox = MakeComboBox(parent, {});
+    QMap<QString, VkPrimitiveTopology>::iterator it;
+    for (it = topologies.begin(); it != topologies.end(); it++) {
+        topologyBox->addItem(it.key());
+    }
```

Comment:
Iterator _it_ doesn't appear to be used outside the for loop, please declare as a for loop variable.

I also recommend looking at range based for loops and preferably using them here, since they have nicer syntax.

`
for (QString& key : topologies.keys()) {
    topologyBox->addItem(key);
}
`

Reply:
Changed this to a range based for loop, not sure why I didn't do that in the first place